pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -0,0 +1,123 @@
+
+  // Geo aggregation functions
+  STUNION("STUnion"),
+
+  // Aggregation functions for multi-valued columns
+  COUNTMV("countMV"),
+  MINMV("minMV"),
+  MAXMV("maxMV"),
+  SUMMV("sumMV"),
+  AVGMV("avgMV"),
+  MINMAXRANGEMV("minMaxRangeMV"),
+  DISTINCTCOUNTMV("distinctCountMV"),
+  DISTINCTCOUNTBITMAPMV("distinctCountBitmapMV"),
+  DISTINCTCOUNTHLLMV("distinctCountHLLMV"),
+  DISTINCTCOUNTRAWHLLMV("distinctCountRawHLLMV"),
+  PERCENTILEMV("percentileMV"),
+  PERCENTILEESTMV("percentileEstMV"),
+  PERCENTILERAWESTMV("percentileRawEstMV"),
+  PERCENTILETDIGESTMV("percentileTDigestMV"),
+  PERCENTILERAWTDIGESTMV("percentileRawTDigestMV"),
+  DISTINCT("distinct");
+
+  private final String _name;
+
+  AggregationFunctionType(String name) {
+    _name = name;
+  }
+
+  public String getName() {
+    return _name;
+  }
+
+  /**
+   * Returns the corresponding aggregation function type for the given function name.
+   * <p>NOTE: Underscores in the function name are ignored.
+   */
+  public static AggregationFunctionType getAggregationFunctionType(String functionName) {
+    String upperCaseFunctionName = StringUtils.remove(functionName, '_').toUpperCase();
+    if (upperCaseFunctionName.startsWith("PERCENTILE")) {
+      String remainingFunctionName = upperCaseFunctionName.substring(10);
+      if (remainingFunctionName.isEmpty() || remainingFunctionName.matches("\\d+")) {
+        return PERCENTILE;
+      } else if (remainingFunctionName.equals("EST") || remainingFunctionName.matches("EST\\d+")) {
+        return PERCENTILEEST;
+      } else if (remainingFunctionName.equals("RAWEST") || remainingFunctionName.matches("RAWEST\\d+")) {
+        return PERCENTILERAWEST;
+      } else if (remainingFunctionName.equals("TDIGEST") || remainingFunctionName.matches("TDIGEST\\d+")) {
+        return PERCENTILETDIGEST;
+      } else if (remainingFunctionName.equals("RAWTDIGEST") || remainingFunctionName.matches("RAWTDIGEST\\d+")) {
+        return PERCENTILERAWTDIGEST;
+      } else if (remainingFunctionName.equals("MV") || remainingFunctionName.matches("\\d+MV")) {
+        return PERCENTILEMV;
+      } else if (remainingFunctionName.equals("ESTMV") || remainingFunctionName.matches("EST\\d+MV")) {
+        return PERCENTILEESTMV;
+      } else if (remainingFunctionName.equals("RAWESTMV") || remainingFunctionName.matches("RAWEST\\d+MV")) {
+        return PERCENTILERAWESTMV;
+      } else if (remainingFunctionName.equals("TDIGESTMV") || remainingFunctionName.matches("TDIGEST\\d+MV")) {
+        return PERCENTILETDIGESTMV;
+      } else if (remainingFunctionName.equals("RAWTDIGESTMV") || remainingFunctionName.matches("RAWTDIGEST\\d+MV")) {
+        return PERCENTILERAWTDIGESTMV;
+      } else {
+        throw new IllegalArgumentException("Invalid aggregation function name: " + functionName);
+      }
+    } else {
+      try {
+        return AggregationFunctionType.valueOf(upperCaseFunctionName);
+      } catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)